ambari-server/src/main/python/bootstrap.py

```diff
@@ -793,25 +798,47 @@ def run(self):
     self.createDoneFile(last_retcode)
     self.status["return_code"] = last_retcode
 
+class ValidateHost(Bootstrap):
+  def __new__(cls, *args, **kwargs):
+    return object.__new__(ValidateHost)
+
+  def __init__(self, hosts, sharedState):
+    super(ValidateHost, self).__init__(hosts, sharedState)
+    self.timeout = HOST_CONNECTIVITY_TIMEOUT
+
+  def login(self):
+    params = self.shared_state
+    self.host_log.write("==========================\n")
+    self.host_log.write("Running login to the host {0} ...".format(self.host))
+    ssh = SSH(params.user, params.sshPort, params.sshkey_file, self.host, "exit",
+              params.bootdir, self.host_log) #login and exit immediately
+    retcode = ssh.run()
+    self.host_log.write("\n")
+    return retcode
 
+  def run(self):
+    self.timeout = HOST_CONNECTIVITY_TIMEOUT
+    self.status["start_time"] = time.time()
+    ret = self.try_to_execute(self.login)
+    retcode = ret["exitstatus"]
+    err_msg = ret["errormsg"]
+    std_out = ret["log"]
+    if retcode != 0:
+      message = "ERROR: Validation of host {0} fails because it finished with non-zero exit code ({1})\nERROR MESSAGE: {2}\nSTDOUT: {3}".format(self.host, retcode, err_msg, std_out)
```

Comment:
Why do we need the word "ERROR" here? Seems like it's just going to be stripped off?